### ISSUE TITLE
Zoom past viewport size

### ISSUE THREAD

**Issue by orbited303** [iina/iina#3208]
First of all, a huge thank-you to everybody involved with the IINA project. It's absolutely fantastic and I can't image my Mac without it. However, my heavy usage has prompted a few ideas:

**What you want IINA to do:**
I would like to be able to zoom into video, preferably in both windowed and full-screen mode (but at least in the latter).

**Now IINA works today:**
When pinching in windowed mode: IINA resizes the window and max out at maximum viewport resolution.
When pinching In full screen mode: Nothing.

**What my use-case is:**
I would like to be able to enlarge (and crop) on video the fly to better be able to see selected details in video. 

**Examples of other projects that have something similar:**
Please see how macOS's Preview does it. If I view a picture in full-screen I can pinch to zoom and then pan (with two fingers) to position viewport over the larger "canvas".

**Comment by TBuTcher71** [comment]
Yes I to need this , This is the last feature holding me into VLC :(

**Comment by holvanato** [comment]
Would love to have this!

**Comment by Muirium** [comment]
Same. This would be fantastic!

**Comment by 7barry** [comment]
Yep

**Comment by 7barry** [comment]
One way to sort of do this is to press cmd+alt+plus repeatedly and it will zoom in your whole OS. I've done this to zoom in further on videos in IINA and it works really well! To zoom back out, just do cmd+alt+minus

**Comment by ItsRyanWu** [comment]
Likewise!

**Comment by svobs** [comment]
Posting this as a sticky note. I stumbled upon [this blog post](https://phoenixtoews.net/blog/2016/3/25/centering-a-resizable-content-view-within-a-nsscrollview-with-swift-22-and-xcode-73) which gives step-by-step instructions for doing what Preview does.

**Comment by robert-corlett** [comment]
Would love to have this! Currently the only way of achieving it is via setting a custom crop size, which is slow and awkward. Movist.app has implemented this feature and it works very nicely. 

The method linked by @svobs above seems like the right way to do it. In the meantime, it looks like a simple solution could be achieved by disabling the check in `handleMagnifyGesture` located in MainWindowController.swift:

```
//Check against max & min threshold
if newHeight < screenFrame.height && newHeight > minSize.height && newWidth > minSize.width {
	let newSize = NSSize(width: newWidth, height: newHeight);
	window.setFrame(window.frame.centeredResize(to: newSize), display: true)
}
```

It checks that the window height doesn't exceed the height of the viewport when using the pinch gesture. However, it doesn't check if the window width exceeds the viewport width, which is why at maximum zoom the window width can exceed the viewport width. If we remove `newHeight < screenFrame.height`, would it give us the behavior we are looking for?

**Comment by zhimaerfan** [comment]
### IINA Version
1.4.x (Release)

### macOS Version
macOS 26.3.1 (Tahoe)

### Description
When zooming into a video, once the zoom level reaches the maximum size of the system viewport (i.e., the video fills the screen area), further zoom operations no longer enlarge the video.

Instead of continuing to zoom in, the image only shifts horizontally (usually to the left). At this point it becomes impossible to zoom further or pan the image to inspect a larger portion of the video. This issue does not exist in Movist; a solution can be sought.

As a result, it is not possible to view enlarged local regions of the video while staying within the maximum system viewport.

### Steps to Reproduce
1. Open any video in IINA.
2. Enter full-screen mode (or a large window).
3. Use pinch gesture or zoom shortcuts to enlarge the video.
4. Continue zooming until the video reaches the system viewport limit.

### Expected Behavior
The video should be able to zoom beyond the viewport size, allowing the user to inspect a magnified portion of the frame.

After zooming beyond the viewport, the user should be able to pan the enlarged image to view different regions of the video (similar to how image viewers handle zoomed images).

### Actual Behavior
Once the video reaches the viewport size limit:
- further zooming does not enlarge the image
- the image only shifts position (typically left)
- panning across the enlarged frame is not possible

### Additional Notes
This behavior prevents users from examining details in videos by zooming and panning within the frame. Other macOS applications (for example Preview when viewing images) allow zooming beyond the viewport and panning across the canvas.

It would be very helpful if IINA could support:
- zooming beyond the viewport size
- free panning of the enlarged video frame

Thank you for your excellent work on IINA.
